Raidboss: Add common replacement text for directions and roles

MaikoTan opened the pull request:
Mentioned in #1821 

- Initial commit: add directions matching regex

- Closes #1821

MaikoTan commented on ui/raidboss/common_replacement.js:
```diff
@@ -9,6 +9,24 @@ const syncKeys = {
+  NW: '(?<= \\(|\\/)NW(?=\\)|\\/)',
+  SE: '(?<= \\(|\\/)SE(?=\\)|\\/)',
+  SW: '(?<= \\(|\\/)SW(?=\\)|\\/)',
+  // Match Roles in replaceText
+  // eg: `(Tank)`, `(Healer)`, `(DPS)`, etc
+  Tank: '(?<= \\(|\\/)Tank(?=\\)|\\/)',
+  Healer: '(?<= \\(|\\/)Healer(?=\\)|\\/)',
+  DPS: '(?<= \\(|\\/)DPS(?=\\)|\\/)',
```
Maybe it is not very useful to translate something like `(T)` and `(T/H)`?
So I have only added full names of roles.

VanSlanzar commented on ui/raidboss/common_replacement.js:
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
+    },
+    [textKeys.Healer]: {
+      de: 'Heiler',
+      fr: 'Soigneurs',
```
Hi, no translate plz ^^. Just `Healer`.

MaikoTan replied:
OK, I have changed it back

Jaehyuk-Lee commented:
Korean east, south, ... it'll be okay to remove `쪽`, it means `-ward`.
ex) `동쪽` is `eastward`. `동` is `east` but it's very unnatural in sentence, this is why translator or many example sentences don't use it alone. People understand its meaning. for reference, `東` is `동`. Same as `東` is `ひがし`. (Korean always write in its pronounce.)

Jaehyuk-Lee commented on ui/raidboss/common_replacement.js:
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
+      fr: 'Tank',
+      ja: 'タンク',
+      cn: '坦克',
+      ko: '방어',
```
ko: '탱커',

Jaehyuk-Lee commented on ui/raidboss/common_replacement.js:
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
+      fr: 'DPS',
+      ja: 'DPS',
+      cn: 'DPS',
+      ko: 'DPS',
```
ko: '딜러',

Jaehyuk-Lee commented on ui/raidboss/common_replacement.js:
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
+      fr: 'E',
+      ja: '東',
+      cn: '东',
+      ko: '동쪽', // FIXME: looks like Korean has no abbrivate of "east"?
```
ko: '동',

Jaehyuk-Lee commented on ui/raidboss/common_replacement.js:
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
+      fr: 'N',
+      ja: '北',
+      cn: '北',
+      ko: '북쪽', // FIXME
```
ko: '북',

Jaehyuk-Lee commented on ui/raidboss/common_replacement.js:
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
+      fr: 'S',
+      ja: '南',
+      cn: '南面',
+      ko: '남쪽', // FIXME
```
ko: '남',

Jaehyuk-Lee commented on ui/raidboss/common_replacement.js:
```diff
@@ -204,6 +222,83 @@ const commonReplacement = {
+      fr: 'O',
+      ja: '西',
+      cn: '西',
+      ko: '서쪽', // FIXME
```
ko: '서',

MaikoTan commented:
Thanks for explaining. I have fixed those words.

MaikoTan commented:
I have changed files with directions that I have added, but I don't know how to deal with the following error.
```
/home/***/workspace/cactbot/ui/raidboss/data/02-arr/raid/t4.js:locale de: pre-translation collision on replaceText '2x Rook (E/W center)' for '\(E/W center\)' and '(?<= \(|\/)E(?=\)|\/)'
/home/***/workspace/cactbot/ui/raidboss/data/02-arr/raid/t4.js:locale de: pre-translation collision on replaceText '2x Rook (E/W center)' for '(?<= \(|\/)E(?=\)|\/)' and '\(E/W center\)'
/home/***/workspace/cactbot/ui/raidboss/data/02-arr/raid/t4.js:locale ja: pre-translation collision on replaceText '2x Rook (E/W center)' for '\(E/W center\)' and '(?<= \(|\/)E(?=\)|\/)'
/home/***/workspace/cactbot/ui/raidboss/data/02-arr/raid/t4.js:locale ja: pre-translation collision on replaceText '2x Rook (E/W center)' for '(?<= \(|\/)E(?=\)|\/)' and '\(E/W center\)'
/home/***/workspace/cactbot/ui/raidboss/data/02-arr/raid/t4.js:locale cn: pre-translation collision on replaceText '2x Rook (E/W center)' for '\(E/W center\)' and '(?<= \(|\/)E(?=\)|\/)'
/home/***/workspace/cactbot/ui/raidboss/data/02-arr/raid/t4.js:locale cn: pre-translation collision on replaceText '2x Rook (E/W center)' for '(?<= \(|\/)E(?=\)|\/)' and '\(E/W center\)'
The terminal process "/bin/bash '-c', 'python test/run_tests.py'" terminated with exit code: 1.
```

The related files are: https://github.com/quisquous/cactbot/blob/d36dd2b545f3e4e9ea87d5b4a001a0f8c0a420e2/ui/raidboss/data/02-arr/raid/t4.txt#L12

https://github.com/quisquous/cactbot/blob/d36dd2b545f3e4e9ea87d5b4a001a0f8c0a420e2/ui/raidboss/data/02-arr/raid/t4.js#L63-L172

Jaehyuk-Lee commented:
The problem is that `(?<= \(|\/)E(?=\)|\/)` catches `E` in `(E/W center)`.
but there's `'\\(E/W center\\)': '(O/W mitte)',` replacement. So, two collides.

You should change
`'\\(E/W center\\)': '(O/W mitte)',` to `'center': 'mitte',`
but there's already replacement for `'\\(center\\)'`, so just remove every `'\\(E/W center\\)'` replacements, then change `'\\(center\\)'` to `center`.

### example
```javascript
      'replaceText': {
        '\\(E/W center\\)': '(O/W mitte)',
        '\\(SE/NW\\)': '(SO/NW)',
        '\\(center\\)': '(mitte)',
        '\\(outside\\)': '(draußen)',
        'Bug': 'Wanze',
        'Dreadnaught': 'Brummonaut',
        'Emergency Override': 'Not-Übersteuerung',
        'Knight': 'Ritter',
        'Rook': 'Drehturm',
        'Soldier': 'Soldat',
      },
```
to
```javascript
      'replaceText': {
        '\\(SE/NW\\)': '(SO/NW)',
        'center': 'mitte',
        '\\(outside\\)': '(draußen)',
        'Bug': 'Wanze',
        'Dreadnaught': 'Brummonaut',
        'Emergency Override': 'Not-Übersteuerung',
        'Knight': 'Ritter',
        'Rook': 'Drehturm',
        'Soldier': 'Soldat',
      },
```

MaikoTan commented:
Thank you! I have fixed it.

quisquous commented:
Thanks for this! This is great.
